pinot-core/src/main/java/org/apache/pinot/core/accounting/PerQueryCPUMemAccountantFactory.java

```diff
@@ -278,8 +278,9 @@ public void updateQueryUsageConcurrently(String queryId) {
      */
     @SuppressWarnings("ConstantConditions")
     public void sampleThreadCPUTime() {
-      if (_isThreadCPUSamplingEnabled) {
-        _threadLocalEntry.get()._currentThreadCPUTimeSampleMS = getThreadResourceUsageProvider().getThreadTimeNs();
+      ThreadResourceUsageProvider provider = getThreadResourceUsageProvider();
+      if (_isThreadCPUSamplingEnabled && provider != null) {
+        _threadLocalEntry.get()._currentThreadCPUTimeSampleMS = provider.getThreadTimeNs();
```

Comment:
Should we add a log if `ThreadResourceUsageProvider` is unexpectedly `null` to help discover execution paths that haven't been instrumented?

Reply:
I thought about that or in some way detect an uninstrumented path. One issue is that this fn. is called for every block of data in every operator. So warn logs can potentially fill up log files.  
Another point - the memory used will be -1 if not instrumented. So this should be good enough _if_ data collection is available.

Reply:
We will also need the operator type to detect an uninstrumented path. I'll add that in a different PR. I am keeping this PR simple to get approval in a short period of time.